solarwinds_apm/configurator.py

```diff
@@ -132,62 +132,70 @@ def _configure_otel_components(
             # before the rest of the processors with defined on_end
             self._configure_txnname_calculator_span_processor(
                 apm_txname_manager,
             )
-            self._configure_inbound_metrics_span_processor(
-                apm_txname_manager,
-                apm_config,
-            )
-            self._configure_otlp_metrics_span_processors(
+
+            if apm_config.get("legacy") is True:
+                # Export APM metrics by APM-proto.
+                self._configure_inbound_metrics_span_processor(
+                    apm_txname_manager,
+                    apm_config,
+                )
+            else:
+                # Export APM metrics, logs by OTLP.
+                # Default values are set by SolarWindsDistro; user can customize.
+                self._configure_metrics_exporter(apm_config)
+                self._configure_otlp_metrics_span_processors(
+                    apm_txname_manager,
+                    apm_config,
+                    oboe_api,
+                )
+                self._configure_logs_exporter(apm_config)
+
+            # Export traces by OTLP or APM-proto depending on OTEL_TRACES_EXPORTER.
+            # Default values are set by SolarWindsDistro; user can customize.
+            self._configure_traces_exporter(
+                reporter,
                 apm_txname_manager,
+                apm_fwkv_manager,
                 apm_config,
-                oboe_api,
             )
+
             # The txnname cleanup span processor must be registered
             # after the rest of the processors with defined on_end
             self._configure_txnname_cleanup_span_processor(
                 apm_txname_manager,
             )
 
-            self._configure_traces_exporter(
-                reporter,
-                apm_txname_manager,
-                apm_fwkv_manager,
-                apm_config,
-            )
-            self._configure_logs_exporter(apm_config)
-            self._configure_propagator()
-            self._configure_response_propagator()
         else:
-            # Warning: This may still set OTEL_PROPAGATORS if set because OTel API
-            logger.error("Tracing disabled. Not setting propagators.")
+            logger.warning(
+                "Tracing disabled. Skipping setup of all telemetry processors and exporters; using no-op tracer."
+            )
+            trace.set_tracer_provider(trace.NoOpTracerProvider())
 
     def _configure_sampler(
         self,
         apm_config: SolarWindsApmConfig,
         reporter: "Reporter",
         oboe_api: "OboeAPI",
     ) -> None:
-        """Always configure SolarWinds OTel sampler, or none if disabled"""
-        if not apm_config.agent_enabled:
-            logger.error("Tracing disabled. Using OTel no-op tracer provider.")
-            trace.set_tracer_provider(trace.NoOpTracerProvider())
-            return
```

Comment:
Removed here and in other helpers because calling `_configure_otel_components` does one enabled/disabled check.